### PR TITLE
Asset Tracker support

### PR DESCRIPTION
**Pull request by avtolstoy**
### Description

Merge of `v1.5.3-tracker.1` to `develop`. This PR also bumps module versions to 2.0.0-alpha.1 and fixes some warnings in system version tests.

### Steps to Test

N/A

### Example App

N/A

### References

N/A

---

### Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: bootloader/src/nRF52840/usbd_device.cpp [hunk 1]
```diff
@@ -146,6 +146,8 @@ class NrfDevice : public Device {
 # define USBD_PRODUCT_NAME              "B SoM"
 #elif PLATFORM_ID == 25
 # define USBD_PRODUCT_NAME              "B5 SoM"
+#elif PLATFORM_ID == 26
+# define USBD_PRODUCT_NAME              "Asset Tracker"
```
Comment (eberseth) [inline bootloader/src/nRF52840/usbd_device.cpp:150]: We need to change "Asset Tracker" to "Tracker" per Ian.

File: build/platform-id.mk [hunk 2]
```diff
@@ -313,6 +317,25 @@ ARM_CPU=cortex-m4
...
+PLATFORM_WIZNET=W5500
+STM32_DEVICE=nRF52840
+PRODUCT_DESC=Production Asset Tracker
+USBD_VID_SPARK=0x2B04
+USBD_PID_DFU=0xD01A
+USBD_PID_CDC=0xC01A
+DEFAULT_PRODUCT_ID=$(PLATFORM_ID)
+PLATFORM_DYNALIB_MODULES=$(PLATFORM_NAME)
+ARM_CPU=cortex-m4
+PLATFORM_THREADING=1
```
Comment (eberseth) [inline build/platform-id.mk:335]: Permanent `SYSTEM_THREAD(ENABLED);`?

Comment (avtolstoy) [reply build/platform-id.mk:335]: No, this just means that the platform supports threading. This is defined for all the other Gen2 and Gen3 platforms as well.

File: bootloader/src/nRF52840/usbd_device.cpp (hunk 1 above)
Comment (avtolstoy) [reply bootloader/src/nRF52840/usbd_device.cpp:150]: @eberseth Can I ask you to confirm what all of these should be:

1. The thing this comment links to which affects the USB string in DFU mode: "%USBD_PRODUCT_NAME DFU Mode"
2. Internal platform product description: https://github.com/particle-iot/device-os/blob/feature/asset-tracker/build/platform-id.mk#L328
3. USB product USB string: https://github.com/particle-iot/device-os/blob/feature/asset-tracker/hal/src/tracker/app_usbd_string_config.h#L85
4. USB Control interface / CDC (serial) USB strings: https://github.com/particle-iot/device-os/blob/feature/asset-tracker/hal/src/tracker/app_usbd_string_config.h#L127
5. How we mention AT in the release list: https://github.com/particle-iot/device-os/blob/feature/asset-tracker/system/system-versions.md

Comment (eberseth) [reply bootloader/src/nRF52840/usbd_device.cpp:150]: 1. "Tracker"
2. "Production Tracker"
3. "Tracker CDC Mode"
4. "Tracker Serial"
5. "Tracker"

Comment (avtolstoy) [reply bootloader/src/nRF52840/usbd_device.cpp]: Fixed.

**Pull request merged**